packages/core/contracts/financial-templates/funding-rate-store/test/MockFundingRateStore.sol

```diff
@@ -19,9 +19,10 @@ contract MockFundingRateStore is FundingRateStoreInterface, Testable {
         // e.g. fundingRate = 0.01 means 1% of token amount charged per second.
         FixedPoint.Signed fundingRate;
         uint256 timestamp; // Time the verified funding rate became available.
+        FixedPoint.Unsigned rewardRate;
     }
 
-    mapping(address => FundingRate[]) private fundingRates;
```

Comment:
We were only using the last index of this FundingRate array for this contract, and the `FundingRateStore` does NOT store an array of FundingRates for each perp.

Reply:
are we sure we want this to be the behavior? does the levered token not perhaps need this info? the downsides for storing past funding rates are simply slightly higher gas, no?

Reply:
Can't we store funding rates just as effectively via events? Also, for the levered token I am pretty sure we can implement it by only knowing the last funding rate.